media/stylus/includes/mixins_indicators.styl

```diff
@@ -0,0 +1,136 @@
+/*
+Mixins for indicators
+====================================================================== */
+$icon-spacing = 30px;
+$inline-horizontal-padding = 0.35em;
+
+/*
+block indicators aka banners
+-------------------------------------------------------------- */
+
+$indicator-block {
+    set-message-base(true);
+    clearfix();
+
+    background-color: $light-background-color;
+    font-size: $smaller-font-size;
+
+    &:before {
+        display: inline-block;
+        width: 30px;
+        bidi-value(float, left, right);
+        bidi-value(margin, -5px 0 -5px ($icon-spacing * -1), -5px ($icon-spacing * -1) -5px 0);
+        font-family: "FontAwesome";
+        font-size: $larger-font-size;
+        font-style: normal;
+        font-weight: normal;
+        text-align: center;
+    }
+
+    pre {
+        color: $text-color;
+    }
+}
+
+$indicator-block-heading {
+    font-size: $base-font-size;
+    font-weight: bold;
+}
+
+$indicator-system {
+    border-width: 2px
+    border-style: solid;
+    font-size: $base-font-size;
+}
+
+/*
+inline indicators aka badges
+-------------------------------------------------------------- */
+
+$indicator-inline {
+    display: inline-block;
+    min-width: 20px;
+    vertical-align: baseline;
+    bidi-style(margin-left, ($grid-spacing / 2 ), margin-right, 0);
+    border-radius: 2px;
+    bidi-style(border-left-width, 4px, border-right-width, 0);
+    bidi-style(border-left-style, solid, border-right-style, none);
+    padding: 0.45em $inline-horizontal-padding 0.30em;
+    background-color: $light-background-color;
+    font-family: $tiny-font-family;
+    font-size: $tiny-font-size;
+    font-weight: normal;
+    letter-spacing: 1px;
+    line-height: normal;
+}
+
+/* icon styling for indicators needs to be seperate because .spec
+   extends $indicator-inline into some :before content */
+$indicator-inline-icon {
+    &:before {
+        display: inline-block;
+        bidi-value(margin-left, ($icon-spacing * -1 + 3px), 3px);
+        bidi-value(margin-right, 3px, ($icon-spacing * -1 + 3px));
+        font-family: 'FontAwesome';
+    }
+}
+
+
+/*
+icons for both indicator types
+-------------------------------------------------------------- */
+
+$indicator-with-icon {
+
+    bidi-style(padding-left, $icon-spacing, padding-right, $inline-horizontal-padding);
+
+    {$selector-icon} {
+        display: none;
+    }
+}
+
+indicator-icon($icon) {
+    @extends $indicator-with-icon;
+
+    &:before {
+        content: $icon;
+    }
+}
+
+/*
+colours for both indicator types
+-------------------------------------------------------------- */
+
+$indicator-text-invert {
+    color: #fff;
+}
+
+$indicator-link-invert {
+    a {
+        color: #ff0;
+    }
+}
+
+$indicator-danger {
+    box-theme($negative);
+}
+
+$indicator-warning {
+    box-theme($warning);
+}
+
+$indicator-info {
+    box-theme($default);
+}
+
+$indicator-obsolete {
+    box-theme($default);
+}
```

Comment:
We could consider separating the `$indicator-info` and `$indicator-obsolete` selectors by a comma, but then again the readability of this is kinda nice, and Stylus is going to compile this anyway. Whatever you think is best. :smile: